backends/ent/backend/retrieve.go

```diff
@@ -0,0 +1,16 @@
+// --------------------------------------------------------------
+// SPDX-FileCopyrightText: Copyright © 2024 The Protobom Authors
+// SPDX-FileType: SOURCE
+// SPDX-License-Identifier: Apache-2.0
+// --------------------------------------------------------------
+package backend
```

Comment:
mmh i'm thinking that the package structure is backwards. I would expect to create a new backend as:


```golang

import "github.com/protobom/storage/backends/ent"

b := ent.New()
```

The current import path ends up being `github.com/protobom/storage/backends/ent/backend`

This could be solved easily by just moving this file one level up, however, at the same time all of the entity clients are at the top:

```golang
NewClient(opts ...Option) *Client
NewDocumentClient(c config) *DocumentClient
NewDocumentTypeClient(c config) *DocumentTypeClient
NewEdgeTypeClient(c config) *EdgeTypeClient
NewExternalReferenceClient(c config) *ExternalReferenceClient
NewHashesEntryClient(c config) *HashesEntryClient
NewIdentifiersEntryClient(c config) *IdentifiersEntryClient
NewMetadataClient(c config) *MetadataClient
NewNodeClient(c config) *NodeClient
NewNodeListClient(c config) *NodeListClient
NewPersonClient(c config) *PersonClient
NewPurposeClient(c config) *PurposeClient
NewToolClient(c config) *ToolClient
```

Do we need these at the top? If the answer is yes, perhaps we could order the code a little bit better my moving the backend constructor method to the top and generating the rest in something like `backend/ent/client/`. I leave it up to you to decide if the generated code would work better in the `client` dir.

As this will be the reference backend, I'm trying to figure out the right ergonomics.

Reply:
Yeah I initially had it in the same package as the rest of the generated code. The reasons I moved it another level deeper were

- I wanted to call it `ent.EntBackend` but the golangci-lint config I use locally, which is pretty strict, said that the imported type stutters
- I took a look at other projects that use ent and this was how it was set up
 
I think swapping the store/retrieve.go and ent generation stuff in the package hierarchy might make sense. I’ll give it a shot

Reply:
@puerco

What do you think about:

- moving the ent code generation stuff into `internal/backends/ent`
- moving `store.go`/`retrieve.go` into `pkg/backends/ent` (or just `backends/ent` if you prefer)?

Reply:
Sounds great, even better since we can control what gets publicly exposed so that people don't take a dependency on it until we're ready.

I think I'd like to keep them as backends/X each of these is more or less a full independent module which we may spin out to its own repo at some point.